### PR TITLE
scripts/create-amis.sh: Update region list

### PR DESCRIPTION
**Pull request by roberth**
## Description of changes

- Re-do #263075
- Fix the command and make the process even easier
   - Removed the excessive indentation. Sorry for the one-time diff, but not adding that to the sed command. Just use the whitespace ~slider~ checkbox in the diff settings this time, or https://github.com/NixOS/nixpkgs/pull/272307/files?diff=split&w=1

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/maintainers/scripts/ec2/create-amis.sh [hunk 1]
```diff
@@ -27,31 +27,37 @@ var ${bucket:=nixos-amis}
 var ${service_role_name:=vmimport}
 
 # Output of the command:
-# > aws ec2 describe-regions --all-regions --query "Regions[].{Name:RegionName}" --output text | sort
+# $ nix-shell -I nixpkgs=. -p awscli --run 'aws ec2 describe-regions --region us-east-1 --all-regions --query "Regions[].{Name:RegionName}" --output text | sort | sed -e s/^/\ \ /'
```
[the remark this answers is not in the document]
Comment (arianvp) [reply nixos/maintainers/scripts/ec2/create-amis.sh:30]: This is because the ec2 api is hosted in a specific region. It's not a global api like iam

**Comment by arianvp** [comment]
I'm currently reworking the upload ami script in https://GitHub.com/arianvp/amis and I will make sure all regions are automatically included there. I beg a bit of patience on that project.

I'm fine with merging this. But we might end up not using this script anymore for next release. Just a heads up.

Still useful change for people uploading their own AMIs though

**Review by arianvp — APPROVED** [review]

File: nixos/maintainers/scripts/ec2/create-amis.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (roberth) [reply nixos/maintainers/scripts/ec2/create-amis.sh:30]: Figured as much.
> somewhat

**Comment by roberth** [comment]
Even better then!

**Pull request merged**